auth/user_mgt.go

```diff
@@ -561,6 +582,261 @@ func (c *baseClient) getUser(ctx context.Context, query *userQuery) (*UserRecord
 	return parsed.Users[0].makeUserRecord()
 }
 
+// A UserIdentifier identifies a user to be looked up.
+type UserIdentifier interface {
+	String() string
+
+	validate() error
+	matchesUserRecord(ur *UserRecord) (bool, error)
+	populateRequest(req *getAccountInfoRequest) error
+}
```

Comment:
Interfaces with many functions (> 2) are a code smell in Go. I think we can push most of this logic onto `getAccountInfoRequest`. See my other comments. Hopefully we can bring it down to just one function: `populateRequest()`

Reply:
(See my other comment for my response.)